Progress bar style fixes + GA events

fairlighteth opened the pull request:
# Summary

<img width="494" alt="Screenshot 2024-08-15 at 15 22 33" src="https://github.com/user-attachments/assets/093556ce-9438-46d6-869d-ab4e4c7016b0">

- Style Safe signing modal
- Fix text auto fit for benefits/surplus display on Finished step
- GA tracking events everywhere and on trigger of each step. Including tracking the time from trigger of the `initial` step to order completion.
- Open console log and search for `CowAnalytics` 
- Extra: Adds a tracking event for closing the CowAMM banner

 
<img width="1018" alt="Screenshot 2024-08-15 at 15 29 07" src="https://github.com/user-attachments/assets/2933b277-1753-410d-b42f-3cdba63b60a2">

elena-zh reviewed (changes requested):
Hey @fairlighteth , great, thank you!

Some issues: 
1. When a surplus is >1USD, the finished modal still does not look good 
![surplus](https://github.com/user-attachments/assets/4c0c3fe7-d595-455e-9abd-905418c2697b)
2. Is this OK that 'expired' and 'finished' states are triggered twice? 
![expired 2 times](https://github.com/user-attachments/assets/2f14d1f9-2f21-40b9-8509-6d38eec06d5c)
3. (not sure) There are no GA events for Discord and 'help' navigation on the Expired step. Should we add them?
![image](https://github.com/user-attachments/assets/232c730e-a13d-4ff7-96e9-f739a9deb919)
4. (same like 3): no GA events for link on every step (placing/solving/etc.). Should we add them? 
![ev](https://github.com/user-attachments/assets/71533a87-8668-4576-912f-f1b1bae6c4e6)
5. There is no event when close the bar (press on the arrow at the top)
6. When I close the progress bar, and then I open Account modal, I can see a bunch of GA events there. Is this OK?
![account](https://github.com/user-attachments/assets/407a2783-08fa-4686-b372-c0347ca6421a)
7. Could you please also check some UI issues I reported https://github.com/cowprotocol/cowswap/pull/4033#issuecomment-2291208722? 

Thanks!

fairlighteth commented:
@elena-zh Addressed all issues that should have been addressed except for https://github.com/cowprotocol/cowswap/pull/4803

elena-zh commented:
@fairlighteth , thanks! But the build has failed. Could you please retrigger it again?

fairlighteth commented:
@elena-zh build should work now

elena-zh commented:
Hey @fairlighteth , thanks for adding GA event and UI updates!
But I've noticed that some new issues with fonts have appeared in the PR: 
1. selected tokens are in bold
![tokens in bold](https://github.com/user-attachments/assets/ceaa0978-4184-47b5-92d6-c46f29090b35)
2. fonts in the token list, swap container (in some fields), in the header differ in comparison with the ones on Prod
![font different](https://github.com/user-attachments/assets/35152abb-f344-4fdb-b0c7-87433661f02b)
![header](https://github.com/user-attachments/assets/7c3d6e9a-8f1a-4994-b1ab-83def0ba18b5)
![Screenshot_10](https://github.com/user-attachments/assets/0f04ef49-493e-4cdf-89d6-a95663f006e6)
3. when opening Account modal I see a lot of analytics requests fired, but there is no active progress bar there
![open a bar](https://github.com/user-attachments/assets/d48cb259-5b18-4ade-94b3-0fefb629d76b)
4. Some dark mode issues are still present: 
![image](https://github.com/user-attachments/assets/913fb9f0-91d4-4feb-9387-59dfe5189dfd)

Could you please take a look at these issues?

fairlighteth commented:
@elena-zh 
1-3: Should be solved
4: The icons are not an issue, it's intended like that. I fixed the opacity on dark mode.

fairlighteth commented:
Oh sorry forgot to mention, I think the plan is to not render this in the activity modal so this should be addressed in another PR.

elena-zh reviewed (approved):
Approved then!

fairlighteth commented:
@elena-zh Also addressed this https://github.com/cowprotocol/cowswap/pull/4803

<img width="549" alt="Screenshot 2024-08-19 at 17 37 30" src="https://github.com/user-attachments/assets/3392f689-ad0b-4c98-af93-777b96bcfd9d">

elena-zh commented:
@fairlighteth , one nitpick: it would be nice to make this message center-aligned in a mobile view: 
![center](https://github.com/user-attachments/assets/a269777d-cc69-4719-a2d0-7cdca8102145)

alfetopito commented on apps/cowswap-frontend/src/legacy/components/TransactionConfirmationModal/DisplayLink.tsx:
```diff
@@ -9,6 +8,7 @@ import { ExternalLinkCustom } from './styled'
 type DisplayLinkProps = {
   id: string | undefined
   chainId: number
+  onClick?: () => void
```
Nitpick: use `Command` type for this.

alfetopito commented on apps/cowswap-frontend/src/modules/wallet/containers/WatchAssetInWallet/index.tsx:
```diff
@@ -18,6 +18,7 @@ export type WatchAssetInWalletProps = {
   shortLabel?: boolean
   className?: string
   fallback?: ReactElement
+  onClick?: () => void
```
Same here: `Command`

fairlighteth commented:
@alfetopito addressed your comments

@elena-zh addressed the Safe tx screen:
<img width="578" alt="Screenshot 2024-08-20 at 12 51 55" src="https://github.com/user-attachments/assets/f73d3125-57b5-4f72-8d33-cab8da059655">
